x/qgb/README.md

```diff
@@ -1,50 +1,320 @@
+  repeated BridgeValidator members = 2 [ (gogoproto.nullable) = false ];
+  // Current chain height
+  uint64 height = 3;
+  // Block time where this valset was created
+  google.protobuf.Timestamp time = 4
+      [ (gogoproto.nullable) = false, (gogoproto.stdtime) = true ];
+}
+```
+
+With `BridgeValidator` representing a validator EVM address and its power:
+
+```protobuf
+// BridgeValidator represents a validator's ETH address and its power
+message BridgeValidator {
+  // Voting power of the validator.
+  uint64 power = 1;
+  // EVM address that will be used by the validator to sign messages.
+  string evm_address = 2;
 }
 ```
 
-- No valset exists in store, so a new valset will be created. This happens mostly after genesis, or after a hard fork.
-- The current block height is the last unbonding height, i.e. when a validator is leaving the validator set. A new valset will need to be created to accommodate that change.
-- A significant power difference happened since the last valset. This could happen if a validator has way more staking power or the opposite. The significant power difference threshold is defined by the constant `SignificantPowerDifferenceThreshold`, and is set to 5% currently.
+1. `nonce`: is the universal nonce of the attestation. It is used to prevent front running attacks in the QGB smart contract. More details can be found in [ADR-004](https://github.com/celestiaorg/celestia-app/blob/main/docs/architecture/adr-004-qgb-relayer-security.md).
+2. `members`: contains the current validator set.
+3. `height`: is the height at which the valset was created.
+4. `time`: is the timestamp of the height at which the valset was created.
 
-#### New data commitment creation
+#### Validator power normalization
 
-A new data commitment is created in the following situation:
+The [QGB bridge power](https://github.com/celestiaorg/celestia-app/blob/6243f26fc419c32940d5dc4eb60b0e0aaf08eaa7/proto/celestia/qgb/v1/types.proto#L12-L13) is obtained by [normalizing](https://github.com/celestiaorg/celestia-app/blob/6243f26fc419c32940d5dc4eb60b0e0aaf08eaa7/x/qgb/keeper/keeper_valset.go#L125-L150) the validators' voting power using the [min-max normalization](https://en.wikipedia.org/wiki/Feature_scaling) formula. This formula takes into account the ratio of each validator's voting power to the total voting power in the block and scales it to a value between `0` and `2^32`.
 
-```golang
-if ctx.BlockHeight() != 0 && ctx.BlockHeight()%int64(k.GetDataCommitmentWindowParam(ctx)) == 0 {
-	...
+By normalizing the voting power, we can significantly reduce the frequency of generating new validator set updates. For example, if there is a small increase in the total on-chain voting power due to inflation, there is no need to create a new validator set. This is because the relative proportions of the validators remain the same, and the normalized QGB power doesn't show any significant difference.
```

Comment:
The percentage change reduces the frequency in case of redelegations for example.
The normalization reduces the frequency in case of, for example, inflation:
```
if there is a small increase in the total on-chain voting power due to inflation, there is no need to create a new validator set. This is because the relative proportions of the validators remain the same, and the normalized QGB power doesn't show any significant difference.
```

Reply:
cool that part makes sense, and including the rational is a good thing.  The part I think we need to change is to also discuss the percentage change, which is the driver of the reduction. We only change if there is a 5% difference. We calculate that difference by first normalizing. This combined mechanism is responsible for safely reducing the frequency of valdiator set updates.